### ISSUE TITLE
Recursive Proof Bug

### ISSUE THREAD

**Issue by sligocki** [sligocki/busy-beaver#4]
The following 10 machines were categorized as halting in < 1000 steps by the `Enumerate.py --recursive`:

```
1RB 1LA  1RC 0LE  1RD 1LC  1LA 0RF  1RD 0LA  1RZ 0RE | None Halt 22 243
1RB 1LA  1RC 0LE  1RD 1LC  1LA 0RF  1LB 0LA  1RZ 0RE | None Halt 22 255
1RB 0LF  1LC 1RB  1LE 0RD  1RC 0RB  1LA 1RE  1RZ 0LD | None Halt 22 259
1RB 0LF  1LC 1RB  1LE 0RD  1LA 0RB  1LA 1RE  1RZ 0LD | None Halt 22 247
1RB 0RE  1LB 1LC  1LD 1RC  1RE 0LF  0RA 1RE  1RZ 1LB | None Halt 48 600
1RB 0LC  1RC 0RE  0LD 1RB  1LB 1LD  1LF 1RE  1RZ 0LA | None Halt 18 550
1RB 0RE  1LB 1LC  1LD 1RC  1LE 0LF  0RA 1RE  1RZ 1LB | None Halt 27 226
1RB 1RC  1RC 0LE  1RD 1LC  1LE 0RF  0LB 1LE  1RZ 1RA | None Halt 38 908
1RB 0LB  1RC 1RB  1RD 1LF  0RE 1LC  1LD 1RE  1RZ 0LA | None Halt 16 185
1RB 0LE  1RC 1LB  1LD 0RF  1RA 1LD  1RD 0LD  1RZ 0RE | None Halt 22 225
```

However, none of these were not classified as halting by the C++ direct simulation code `lr_enum` which was run for 1000 steps.

Looking at just the first machine: `Quick_Sim.py -rn1` shows it halting at step 243! But `Quick_Sim.py -rn2` shows it infinite ...

**Comment by sligocki** [comment]
I've tracked this bug down. It's a bug in minimum checking for tape rep counts for recursive rules. When we apply a child rule inside the recursive rule, we are not correctly enforcing the minimums from the child rule into the recursive context.

**Comment by sligocki** [comment]
Fixed by b89deffeb1240cbe58f662482405da7b2bd32649
